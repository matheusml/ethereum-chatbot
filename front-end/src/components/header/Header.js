import React from "react";
import { View, Text, StyleSheet } from "react-native";
import PropTypes from "prop-types";

export default class Header extends React.Component {
  render() {
    return (
      <View style={styles.header}>
        <Text style={styles.title}>#{this.props.title}</Text>
      </View>
    );
  }
}

Header.defaultProps = {
  title: "Channel"
};

Header.propTypes = {
  title: PropTypes.string
};

const styles = StyleSheet.create({
  header: {
    height: 80,
    backgroundColor: "#0aa6dd",
    alignItems: "center",
    justifyContent: "flex-end",
    padding: 10
  },
  title: {
    color: "white",
    fontWeight: "bold",
    fontSize: 24
  }
});
