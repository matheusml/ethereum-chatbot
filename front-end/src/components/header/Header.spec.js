import React from "react";

import Header from "./Header";

import renderer from "react-test-renderer";

const props = {
  title: "channel title"
};

describe("Header", () => {
  it("should match snapshot without props", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot with props", () => {
    const tree = renderer.create(<Header {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
