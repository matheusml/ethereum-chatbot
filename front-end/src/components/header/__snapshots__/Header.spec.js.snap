// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot with props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "lightseagreen",
      "height": 80,
      "justifyContent": "flex-end",
      "padding": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    #
    channel title
  </Text>
</View>
`;

exports[`Header should match snapshot without props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "lightseagreen",
      "height": 80,
      "justifyContent": "flex-end",
      "padding": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    #
    Channel
  </Text>
</View>
`;
